imports/database/database-access-object.js

```diff
@@ -0,0 +1,37 @@
+export class DatabaseAccessObject {
+    constructor(collection) {
+        this._collection = collection;
+    }
+
+    find(queryObject) {
+        return this._collection.find(queryObject);
+    }
+
+    findOne(queryObject) {
+        return this._collection.findOne(queryObject);
+    }
+
+    insert(object) {
+        this._collection.schema.validate(object);
```

Comment:
`.validate(object)` will throw an exception, which we definitely want to propagate further up. The DAO is a flexible class that can be used in a variety of contexts and its up to the user to determine how it should handle error handling.